applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp

```diff
@@ -24,368 +24,272 @@
+{
+    const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+    Matrix F(dim,dim);
+    
+    if(dim == 2) {
+        F(0,0) = 1.0+rStrainTensor(0);  
+        F(0,1) = 0.5*rStrainTensor(2); 
+        F(1,0) = 0.5*rStrainTensor(2);   
+        F(1,1) = 1.0+rStrainTensor(1);
+    } else {
+        F(0,0) = 1.0+rStrainTensor(0);     
+        F(0,1) = 0.5*rStrainTensor(3); 
+        F(0,2) = 0.5*rStrainTensor(5);
+        F(1,0) = 0.5*rStrainTensor(3);   
+        F(1,1) = 1.0+rStrainTensor(1);   
+        F(1,2) = 0.5*rStrainTensor(4);
+        F(2,0) = 0.5*rStrainTensor(5);   
+        F(2,1) = 0.5*rStrainTensor(4); 
+        F(2,2) = 1.0+rStrainTensor(2);
+    }
+    
+    return F;
+}
 
-        // Verify that the constitutive law exists
-        if ( this->GetProperties().Has( CONSTITUTIVE_LAW ) == false )
-        {
-            KRATOS_ERROR << "Constitutive law not provided for property " << this->GetProperties().Id() << std::endl;
-        }
+/***********************************************************************************/
+/***********************************************************************************/
 
+int  SmallDisplacement::Check( const ProcessInfo& rCurrentProcessInfo )
+{
+    KRATOS_TRY
 
-        // Verify that the constitutive law has the correct dimension
-        if ( dimension == 2 )
-        {
-//             if ( this->GetProperties().Has( THICKNESS ) == false ) // NOTE: Not mandatory
-//             {
-//                 KRATOS_ERROR << "THICKNESS not provided for element " << this->Id() << std::endl;
-//             }
+    int ier = BaseSolidElement::Check(rCurrentProcessInfo);
 
-            if ( this->GetProperties().GetValue( CONSTITUTIVE_LAW )->GetStrainSize() < 3 || this->GetProperties().GetValue( CONSTITUTIVE_LAW )->GetStrainSize() > 4 )
-            {
-                KRATOS_ERROR << "Wrong constitutive law used. This is a 2D element! expected strain size is 3 or 4 (el id = ) " << this->Id() << std::endl;
-            }
-        }
-        else
-        {
-            if ( this->GetProperties().GetValue( CONSTITUTIVE_LAW )->GetStrainSize() != 6 )
-            {
-                KRATOS_ERROR << "Wrong constitutive law used. This is a 3D element! expected strain size is 6 (el id = ) " << this->Id() << std::endl;
-            }
-        }
+    return ier;
 
-        //check constitutive law
-        for ( unsigned int i = 0; i < mConstitutiveLawVector.size(); i++ )
-        {
-            return mConstitutiveLawVector[i]->Check( GetProperties(), GetGeometry(), rCurrentProcessInfo );
-        }
+    KRATOS_CATCH( "" );
+}
 
-        //check if it is in the XY plane for 2D case
+/***********************************************************************************/
+/***********************************************************************************/
 
-        return 0;
+void SmallDisplacement::save( Serializer& rSerializer ) const
+{
+    rSerializer.save( "Name", "SmallDisplacement" );
```

Comment:
I am pretty sure that this line has to be removed

Reply:
yep, my fault